src/db.rs

```diff
@@ -501,18 +501,23 @@ impl Database {
         options: RunCursorCommandOptions,
         session: &mut ClientSession,
     ) -> Result<SessionCursor<Document>> {
+        let mut selection_criteria = SelectionCriteria::ReadPreference(options.read_preference.clone().unwrap()).into();
```

Comment:
Breaking down what's going on here:
* `options.read_preference.clone()` - creates a temporary value, of type `Option<ReadPreference>`.  Not using `clone()` here would move the value of `read_preference` out of `options`, leaving the latter partially-initialized, so yup, makes sense.
* `<above>.unwrap()` - this consumes the temporary `Option<ReadPreference>` and produces a `ReadPreference`.  It does this by checking whether the value is `Some` (it can just return the value) or `None` (it terminates the program with a panic).  Since it would be quite impolite of us to crash the user's program, using `unwrap` is something we avoid in almost all circumstances.
* `SelectionCriteria::ReadPreference(<above>)` - straightforward.
* `<above>.into()` - the Rust standard library provides the `Into` trait (and its mirror `From`) for general type conversions.  What it actually does depends on both the type being converted from and the type being converted to.  In this case, the compiler can tell that the source type is `SelectionCriteria` (because you just constructed it and are calling `into()` directly on it) and can infer that the destination type is `Option<SelectionCriteria>` (because you `match` against it below with `Some()` and `None`).  Converting some value `v` of type `T` into a new value of type `Option<T>` is very simple - it just constructs `Some(v)`.

So, putting that all together, what this line will do:
* if `options.read_preference` contains a `Some(v)` value, this will construct a `Some(SelectionCriteria::ReadPreference(v))` - yup, that's what we want 👍 
* if `options.read_preference` contains a `None` value, this will crash! 👎 

What we *actually* want is to turn a `Some(v)` into a `Some(SelectionCriteria::ReadPreference(v))` and a `None` into, well, `None`.  This is an excellent place for a `match`, since that's exactly what match is for 🙂:
```
let mut selection_criteria = match options.read_preference.clone() {
  Some(v) => Some(SelectionCriteria::ReadPreference(v)),
  None => None,
};
```

It turns out that with `Option` this kind of "wrap it if it's `Some`, do nothing if it's `None`" pattern is _very_ common, so there's a standard library function to do just that, [`Option::map`](https://doc.rust-lang.org/std/option/enum.Option.html#method.map).  Using that, this becomes:
```
let mut selection_criteria = options.read_preference
    .clone()
    .map(|v| SelectionCriteria::ReadPreference(v));
```
As a bit of a side note, because the closure `|v| SelectionCriteria::ReadPreference(v)` is just passing its argument directly through, this can be simplified slightly:
```
let mut selection_criteria = options.read_preference
    .clone()
    .map(SelectionCriteria::ReadPreference);
```

Working with values like this can definitely feel a bit foreign if you're not used to languages that use _algebraic data types_ as a core feature.  I'd encourage you to approach situations like this by starting with a "what data do I have, and what data do I need" analysis, seeing if you can translate that into a `match` statement (or multiple, if needed), and then looking to see if there are utility functions for what you've ended up doing.